auth/README.md

```diff
@@ -146,6 +147,67 @@ allprojects {
 }
 ```
 
+#### GitHub
+
+WARNING: GitHub OAuth is not for the faint of heart. Getting it setup correctly is an invested
+process and may take a half-hour or two. Ready? Let's begin.
+
+##### Adding secrets
+
+Hop over to your [GitHub app](https://github.com/settings/developers) and grab its client ID and
+Secret to put them in your resources:
+
+```xml
+<resources>
+    <string name="github_client_id" translatable="false">YOUR_CLIENT_ID</string>
+    <string name="github_client_secret" translatable="false">YOUR_CLIENT_SECRET</string>
+</resources>
+```
+
+##### Adding your Firebase web host
+
+Next, find your project id in the Firebase Console's project settings and add it to your resources
+in the form `project-id.firebaseapp.com`:
+
+```xml
+<resources>
+    <string name="firebase_web_host" translatable="false">project-id.firebaseapp.com</string>
+</resources>
+```
+
+##### Getting a SHA-256 hash of your keystore
+
+[Run the `keytool` utility](https://developers.google.com/android/guides/client-auth) found in your
+JDK's installation folder to get a SHA-256 hash of your release keystore.
+
+Protip: you might as well also grab the release SHA-1 hash and the debug hashes to add them to the
+Firebase Console since they're useful in other contexts. Also, adding debug hashes will let you test
+all this without having to use a release build.
+
+##### Deploying a Firebase Hosting solution
+
+If you're already using Firebase Hosting, give yourself a pat on the back and move on. Otherwise,
+tough luck. 🤷‍♂️ Go through [this tutorial](https://firebase.google.com/docs/hosting/quickstart) and
+make sure to say no when asked to
+[redirect](https://firebase.google.com/docs/hosting/url-redirects-rewrites) everything to a single
+page. If you're already doing that, exclude `.well-known/assetlinks.json`.
+
+##### Adding asset links
+
+So close, you're almost there! Follow step 1 of
+[this tutorial](https://developers.google.com/digital-asset-links/v1/getting-started#quick-usage-example)
```

Comment:
Where we can, let's inline information rather than linking out.  For example we can provide the format of the `json` file:

```json
[{
  "relation": ["delegate_permission/common.handle_all_urls"],
  "target" : {
    "namespace": "android_app",
    "package_name": "com.your.package.name",
    "sha256_cert_fingerprints": ["your_sha_256_fingerprint"]
  }
}]
```